tools/clean/provider/gcp.go

```diff
@@ -102,60 +102,146 @@ func (gcp *GCP) DeletePrivateEndpoint(ctx context.Context, groupName, sAttachmen
 	return nil
 }
 
-func (gcp *GCP) DeleteOrphanPrivateEndpoints(ctx context.Context, lifetime int, region string, subnet string) ([]string, []string, error) {
+func (gcp *GCP) DeleteOrphanVPCs(ctx context.Context, lifetimeHours int, vpcNamePrefix string) ([]string, []string, []error) {
+	vpcs := gcp.networkClient.List(ctx, &computepb.ListNetworksRequest{
+		Project: gcp.projectID,
+	})
+
+	var done, skipped []string
+	var errs []error
```

Comment:
OOC, is there a (realistic) difference between doing this and using `errors.Join`?

Reply:
the semantics are different, `errors.Join` returns one error, as a composite of multiple ones. Here, we want to access every single error later for rendering, that is not possible with an error that was joined.